compiler/noirc_frontend/src/elaborator/statements.rs

```diff
@@ -435,7 +437,19 @@ impl<'context> Elaborator<'context> {
         None
     }
 
-    pub(super) fn elaborate_comptime(&self, _statement: Statement) -> (HirStatement, Type) {
-        todo!("Comptime scanning")
+    pub(super) fn elaborate_comptime(&mut self, statement: Statement) -> (HirStatement, Type) {
+        let span = statement.span;
+        let (hir_statement, typ) = self.elaborate_statement(statement);
+        let mut interpreter = Interpreter::new(self.interner, &mut self.comptime_scopes);
+
+        if let Err(error) = interpreter.evaluate_statement(hir_statement) {
+            self.errors.push(error.into_compilation_error_pair());
+        }
+
+        let unit = HirExpression::Literal(HirLiteral::Unit);
```

Comment:
I'm not clear on why we're returning a unit value rather than the evaluation of `hit_statement`.

Reply:
Hmm so this was meant to remove the `comptime` statements from the runtime program by replacing them with unit values (thinking because statements always(*) return unit) but I forgot about the `StatementKind::Expression` case and I think the case of inlining macros as well. Good catch, I'll push a fix.

Reply:
Should be fixed now.